roles/openshift_master/tasks/main.yml

```diff
@@ -171,6 +171,14 @@
   when: openshift_master_ha | bool and openshift_master_cluster_method == "native"
 # end workaround for missing systemd unit files
 
+# If the session secrets file was missing from any master, we should
+# clean up the file on masters so that we can get them in sync.
+- name: Delete session secrets file
+  file:
+    path: "{{ openshift.master.session_secrets_file }}"
+    state: absent
+  when: session_secrets_missing | bool
+
 - name: Create session secrets file
   template:
```

Comment:
The `force: no` here should have been preventing us from overwriting the session secrets anyway, so the idempotency issues would be limited to the commands that are being run to set the variables...  That said, we don't want to reset those variables, since they will update the values stored by openshift_facts which will no longer match the files on disk.